search/2016-09-01/swagger/searchservice.json

```diff
@@ -0,0 +1,3797 @@
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/IndexerExecutionInfo"
+            }
+          }
+        }
+      }
+    },
+    "/indexes": {
+      "post": {
+        "tags": [
+          "Indexes"
+        ],
+        "operationId": "Indexes_Create",
+        "description": "Creates a new Azure Search index.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/rest/api/searchservice/Create-Index"
+        },
+        "parameters": [
+          {
+            "name": "index",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Index"
+            },
+            "description": "The definition of the index to create."
+          },
+          {
+            "name": "client-request-id",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "format": "uuid",
+            "description": "The tracking ID sent with the request to help with debugging.",
+            "x-ms-client-request-id": true,
+            "x-ms-parameter-grouping": { "name": "search-request-options" }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "x-ms-request-id": "request-id",
+        "responses": {
+          "201": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/Index"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "Indexes"
+        ],
+        "operationId": "Indexes_List",
+        "description": "Lists all indexes available for an Azure Search service.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/rest/api/searchservice/List-Indexes"
+        },
+        "parameters": [
+          {
+            "name": "$select",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Selects which properties of the index definitions to retrieve. Specified as a comma-separated list of JSON property names, or '*' for all properties. The default is all properties."
+          },
+          {
+            "name": "client-request-id",
+            "in": "header",
+            "required": false,
+            "type": "string",
+            "format": "uuid",
+            "description": "The tracking ID sent with the request to help with debugging.",
+            "x-ms-client-request-id": true,
+            "x-ms-parameter-grouping": { "name": "search-request-options" }
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "x-ms-request-id": "request-id",
+        "responses": {
+          "200": {
+            "description": "",
+            "schema": {
+              "$ref": "#/definitions/IndexListResult"
```

Comment:
One thing I noticed:
```
IndexListResult
IndexerListResult
DocumentIndexResult
DataSourceListResult
```
merely encapsulate an array `value` (the actual result).
Consider marking operations that return one of the above as
```
"x-ms-pageable": {
  "nextLinkName": null
}
```
This will make the encapsulated list the actual return type (e.g. `IEnumerable<Indexer>` instead of `IndexerListResult`) - in case that is something you desire.